sdcv.el

```diff
@@ -603,28 +600,21 @@ Argument DICTIONARY-LIST the word that need transform."
 (defun sdcv-translate-result (word dictionary-list)
   "Call sdcv to search word in dictionary list, return filtered
 string of results."
-  (sdcv-filter
-   (shell-command-to-string
+  (with-temp-buffer
     ;; Set LANG environment variable, make sure `shell-command-to-string' can handle CJK character correctly.
-    (format "env LANG=%s %s -x -n %s %s --data-dir=%s"
-            sdcv-env-lang
-            sdcv-program
-            (mapconcat (lambda (dict)
-                         (concat "-u \"" dict "\""))
-                       dictionary-list " ")
-            (format "\"%s\"" word)
-            sdcv-dictionary-data-dir))))
-
-(defun sdcv-filter (sdcv-string)
-  "This function is for filter sdcv output string,.
-Argument SDCV-STRING the search string from sdcv."
-  (setq sdcv-string (replace-regexp-in-string sdcv-filter-string "" sdcv-string))
-  (if (equal sdcv-string "")
-      sdcv-fail-notify-string
-    (with-temp-buffer
-      (insert sdcv-string)
+    (let ((process-environment (cons (concat "LANG=" sdcv-env-lang) process-environment)))
+      (apply #'call-process sdcv-program nil t nil
+             "-x" "-n" "--data-dir" sdcv-dictionary-data-dir
```

Comment:
The `word` arg is missing, maybe change to something like this:

```suggestion
    (let ((process-environment (cons (concat "LANG=" sdcv-env-lang) process-environment))
		  (word (format "\"%s\"" word)))
      (apply #'call-process sdcv-program nil t nil
             "-x" "-n" word "--data-dir" sdcv-dictionary-data-dir
```

Reply:
Ok I test this PR with a MS-windows machine, here is what I found:

With the patch I provided, is works almost fine, except the IPA string are not correctly renderd, those are unicode charactors but in the output buffer they are renderd as `\u02d0m` or something similar.

Change `sdcv-env-lang` to **whatever** LANG won't change anything, dose this means the program dose not cares about `process-environment` ?  I should mention that sdcv never formally ported to windows, windows users have to compile and run sdcv under MSYS2 environment.

So I tried  add the flag `--utf8-output` but it make things worse, now the entire buffer is gibberish...  🤣

Reply:
> Change `sdcv-env-lang` to **whatever** LANG won't change anything, dose this means the program dose not cares about `process-environment` ? I should mention that sdcv never formally ported to windows, windows users have to compile and run sdcv under MSYS2 environment.

IDK, in theory `process-environment` works exactly like `env` utility, but without unnecessary `exec`s.

Reply:
Its the sdcv program's problem I think. On MS-windows, run sdcv command in a MSYS2 shell, the program always output the result encoded according to the OS default locale, whatever the LANG variable in the shell is, unless the `--utf8-output`
flag is used. 

With the 'shell-command-to-string' method add this `--utf8-output` flag can gives me perfect output tho...

Reply:
I think on windows, the user, can always set the command, explicitly, to `("env" "LANG=lang" "sdcv" ...etc..)`